### PR TITLE
Add k8s.{pod,node}.network.{io,errors} metrics

### PR DESCRIPTION
**Pull request by ChrsMark**
Part of https://github.com/open-telemetry/semantic-conventions/issues/1032

Fixes https://github.com/open-telemetry/semantic-conventions/issues/1487

## Changes

This PR adds the following metrics:
- `k8s.pod.network.io`
- `k8s.pod.network.errors`
- `k8s.node.network.io`
- `k8s.node.network.errors`

The above metrics are already used in the Collector in the [kubeletstats](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) receiver and the naming is aligned with the [System metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/model/system/metrics.yaml).

These metrics come with the following attributes `network.io.direction` and `system.device` which are already part of SemConv registry and are used in the [System metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/model/system/metrics.yaml). The attributes in the Collector will need to be adjusted to follow the recent SemConv rules.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: model/k8s/metrics.yaml
Comment (TylerHelmuth) [inline model/k8s/metrics.yaml:42]: Is this the equivalent of `interface`?

File: docs/system/k8s-metrics.md
Comment (TylerHelmuth) [inline docs/system/k8s-metrics.md:111]: To save ourselves work later and to start preparing end users, can we start a list somewhere, maybe in the readme, of the breaking changes that will be included in the stable versions of these semantics?

I know this particular change is additive in this repo, but considering that the kubeletstatsreceiver has been emitting this metric with the attributes `interface` and `direction` since its inception this change will be very impactful.

File: model/k8s/metrics.yaml
Comment (ChrsMark) [reply model/k8s/metrics.yaml:42]: Yes, that is designed based on the [hostmetricsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/internal/scraper/networkscraper/documentation.md#systemnetworkio) which will also need to be updated to the latest SemConv once the System metrics are stable.

File: docs/system/k8s-metrics.md
Comment (ChrsMark) [reply docs/system/k8s-metrics.md:111]: Good idea. I wonder where this information would fit best. @open-telemetry/specs-semconv-maintainers @open-telemetry/specs-semconv-approvers any suggestions?

Comment (jsuereth) [reply docs/system/k8s-metrics.md:111]: I think, ideally, we should be able to understand breaking changes and document them in Schema URL eventually for this purpose.

We don't have a good mechanism right now to do that, but @trask and @lmolkova have the most experience having stabilzied HTTP and soon, DB.

Comment (ChrsMark) [reply docs/system/k8s-metrics.md:111]: One thing we can do for the Collector specifically is to file a meta issue that will collect all this breaking changes.

Additionally we could update the k8s' components that will be affected to have their documentation explicitly mentioning this breaking change meta issue.

File: model/k8s/metrics.yaml
Comment (ChrsMark) [reply model/k8s/metrics.yaml:42]: However, I'm not really sure if that's a good and settled decision right now: https://github.com/open-telemetry/semantic-conventions/issues/308#issuecomment-2391638296

File: docs/system/k8s-metrics.md
Comment (lmolkova) [reply docs/system/k8s-metrics.md:111]: we usually have a migration plan as a part of stabilization process - like this one 

https://github.com/open-telemetry/semantic-conventions/blob/cf4ce09650a7a8bafc41a13d062e6f8ac64ade84/docs/database/README.md?plain=1#L18-L39

In a few words 
- the instrumentation should keep doing what it's doing now (following semconv or not)
- it should have an option to enable new conventions (for collector I'd assume it not be env var, but it'd make sense to have consistent way of opting in across different receivers). 
- it can drop old conventions or just switch to a new ones by default with the next major version. 

We've been also creating migration guides like these https://github.com/open-telemetry/semantic-conventions/blob/main/docs/non-normative/http-migration.md or https://github.com/open-telemetry/semantic-conventions/blob/main/docs/non-normative/db-migration.md which outline the list of changes. 

You can probably do something similar and just mention undocumented (in semconv) old attributes.

File: model/k8s/metrics.yaml
Comment (lmolkova) [inline model/k8s/metrics.yaml:91]: could you please elaborate? Is it a number of bytes transmitted by the node?

File: model/k8s/metrics.yaml
Comment (lmolkova) [inline model/k8s/metrics.yaml:101]: Could you elaborate on this one? Is there some external documentation we can link, what kinds of errors does it cover?

File: model/k8s/metrics.yaml
Comment (lmolkova) [inline model/k8s/metrics.yaml:106]: is there any information about the error we can include? We have a very vague `error.type` attribute intended for this purpose.

File: model/k8s/metrics.yaml
Comment (lmolkova) [inline model/k8s/metrics.yaml:48]: similar to the node errors - could you please elaborate on what it means?

File: model/k8s/metrics.yaml
Comment (lmolkova) [inline model/k8s/metrics.yaml:53]: can we report any meaningful `error.type` here?

File: model/k8s/metrics.yaml
Comment (lmolkova) [inline model/k8s/metrics.yaml:38]:
```suggestion
    brief: "Number of bytes transmitted by the pod"?
```

Comment (ChrsMark) [reply model/k8s/metrics.yaml:38]: It's not only transmitted but also received. That's why I guess we used `IO` in the kubeletstats' docs.

File: model/k8s/metrics.yaml
Comment (ChrsMark) [reply model/k8s/metrics.yaml:101]: The best external reference we can find is from the Kubelet's stats API: https://github.com/kubernetes/kubernetes/blob/v1.31.1/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go#L189-L204

File: model/k8s/metrics.yaml
Comment (ChrsMark) [reply model/k8s/metrics.yaml:106]: I don't think we have any information about the errors' types. It seems these are aggregated errors over a specific interface: https://github.com/kubernetes/kubernetes/blob/v1.31.1/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go#L189-L204

File: docs/system/k8s-metrics.md
Comment (ChrsMark) [reply docs/system/k8s-metrics.md:111]: Thank's for the information @lmolkova! If I understand correctly could we create a similar migration plan file for K8s already and start collecting the diffs there?

If I understood @TylerHelmuth initial comment here, we would like to start informing users already about such changes/diffs even if we are far from stability. In this way users could already be prepared about what changes to expect.

File: model/k8s/metrics.yaml
Comment (lmolkova) [reply model/k8s/metrics.yaml:38]: Makes sense! Is there something more elaborate we can say than "Pod network IO" ?

Comment (ChrsMark) [reply model/k8s/metrics.yaml]: I made it `Network bytes for the Pod`. Guess it's more descriptive?

File: model/k8s/metrics.yaml
Comment (ChrsMark) [reply model/k8s/metrics.yaml:42]: If https://github.com/open-telemetry/semantic-conventions/pull/1492 makes it, we will use `network.interface.name` here.

Comment (ChrsMark) [reply model/k8s/metrics.yaml]: Changed to `network.interface.name`: [804d91a](https://github.com/open-telemetry/semantic-conventions/pull/1427/commits/804d91a851a0184d830b881beb47ac2b95e2d66a)

File: docs/system/k8s-metrics.md
Comment (TylerHelmuth) [reply docs/system/k8s-metrics.md]: @joaopgrassi I love it, thats exactly what I want to do here

**Comment by ChrsMark** [comment]
@TylerHelmuth @lmolkova anything blocking this one?

File: docs/system/k8s-metrics.md
Comment (TylerHelmuth) [reply docs/system/k8s-metrics.md]: @ChrsMark where did we land with trying to track these breaking changes as we go?

Comment (ChrsMark) [reply docs/system/k8s-metrics.md]: As mentioned at https://github.com/open-telemetry/semantic-conventions/pull/1427#issuecomment-2449304586, I have filed https://github.com/open-telemetry/semantic-conventions/issues/1537 for the migration plan (following the idea suggested above). I will start working on the migration plan PR soonish.

**Review by TylerHelmuth — APPROVED** [review]

File: model/k8s/metrics.yaml
Comment (ChrsMark) [reply model/k8s/metrics.yaml]: Seems, this is covered. I'm going to resolve this conversation.

File: docs/system/k8s-metrics.md
Comment (ChrsMark) [reply docs/system/k8s-metrics.md]: Seems we are covered here. I'm going to resolve this conversation.

**Review by lmolkova — APPROVED** [review]

**Comment by lmolkova** [comment]
@ChrsMark please regenerate markdown files (there was a small change on main) and it should be ready to go.

**Pull request merged**
